src/factory/CirclesBackingFactory.sol

```diff
@@ -0,0 +1,365 @@
+        //if (inflationaryCircles == address(0)) revert InflationaryCirclesNotExists(avatar);
+    }
+
+    // Internal functions
+
+    // deploy instance
+    /**
+     * @notice Deploys a new CirclesBacking contract with CREATE2.
+     * @param backer Address which is backing circles.
+     * @return deployedAddress Address of the deployed contract.
+     */
+    function deployCirclesBacking(address backer) internal returns (address deployedAddress) {
+        // open question: do we want backer to be able to create only one backing? - this is how it is now.
+        // or we allow backer to create multiple backings, 1 per supported backing asset - need to add backing asset to salt.
+        bytes32 salt_ = keccak256(abi.encodePacked(backer));
+
+        deployedAddress = address(new CirclesBacking{salt: salt_}());
+
+        if (deployedAddress == address(0) || deployedAddress.code.length == 0) {
+            revert CirclesBackingDeploymentFailed(backer);
+        }
+
+        // link instance to backer
+        backerOf[deployedAddress] = backer;
+
+        emit CirclesBackingDeployed(backer, deployedAddress);
+    }
+
+    // cowswap app data helper
+    /// @dev returns string as address value
+    function addressToString(address _addr) internal pure returns (string memory) {
+        bytes32 value = bytes32(uint256(uint160(_addr)));
+        bytes memory alphabet = "0123456789abcdef";
+
+        bytes memory str = new bytes(42);
+        str[0] = "0";
+        str[1] = "x";
+
+        for (uint256 i = 0; i < 20; i++) {
+            str[2 + i * 2] = alphabet[uint8(value[i + 12] >> 4)];
+            str[3 + i * 2] = alphabet[uint8(value[i + 12] & 0x0f)];
+        }
+
+        return string(str);
+    }
+
+    // personal circles lbp name
+    function _name(address inflationaryCirlces) internal view returns (string memory) {
+        return string(abi.encodePacked(LBP_PREFIX, IERC20Metadata(inflationaryCirlces).name()));
+    }
+
+    // personal circles lbp symbol
+    function _symbol(address inflationaryCirlces) internal view returns (string memory) {
+        return string(abi.encodePacked(LBP_PREFIX, IERC20Metadata(inflationaryCirlces).symbol()));
+    }
+
+    // Callback
+    function onERC1155Received(address operator, address from, uint256 id, uint256 value, bytes calldata data)
+        external
+        returns (bytes4)
+    {
+        if (msg.sender != address(HUB_V2)) revert OnlyHub();
+        if (value != CRC_AMOUNT) revert NotExactlyRequiredCRCAmount(CRC_AMOUNT, value);
+        address avatar = address(uint160(id));
+        if (operator != from || from != avatar) revert BackingInFavorDissalowed();
+        // handling personal CRC
+        // get stable address
+        address stableCRC = getPersonalCircles(avatar);
+
+        uint256 stableCirclesAmount = IERC20(stableCRC).balanceOf(address(this));
+        // wrap erc1155 into stable ERC20
+        HUB_V2.wrap(avatar, CRC_AMOUNT, uint8(1));
```

Comment:
`hub.wrap` will return the address of the stable erc20 where the tokens get wrapped. So you can write

```
address stableCRC = HUB_V2.wrap(avatar, CRC_AMOUNT, uint8(1));
```
and safely remove lines 352 and 354

Reply:
ignore this, because you need to get the balance before wrapping; but the `LIFT_ERC20.ensureERC20()` will still help